consensus/bor/bor.go

```diff
@@ -349,8 +349,9 @@ func (c *Bor) verifyHeader(chain consensus.ChainHeaderReader, header *types.Head
 
 	number := header.Number.Uint64()
 
-	// Don't waste time checking blocks from the future
-	if header.Time > uint64(time.Now().Unix()) {
+	// Don't waste time checking blocks from the future but allow a buffer of block time for
+	// early block announcements.
+	if header.Time-c.config.CalculatePeriod(number) > uint64(time.Now().Unix()) {
```

Comment:
I believe we should check if the validator is primary here or not. 

Because with `period` set to 2s, non primary validators could also send blocks 2s early than their expected time. Wdyt?

Reply:
Will need access to snapshot (the one used in bor consensus) to find out the succession number of proposer and it's fetched very late in this `verifyHeader` call so we'll probably have to move this check after that instead of keeping it here.